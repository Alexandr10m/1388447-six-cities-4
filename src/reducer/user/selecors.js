import NameSpace from "../name-space.js";


const getAuthorizationStatus = (state) => state[NameSpace.USER].authorizationStatus;


export {getAuthorizationStatus};
