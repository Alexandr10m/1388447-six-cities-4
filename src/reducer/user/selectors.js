import NameSpace from "../name-space.js";


const USER = NameSpace.USER;

const getAuthorizationStatus = (state) => state[USER].authorizationStatus;
const getAuthInfo = (state) => state[USER].authInfo;
const getEmail = (state) => state[USER].authInfo.email || null;


export {getAuthorizationStatus, getAuthInfo, getEmail};
