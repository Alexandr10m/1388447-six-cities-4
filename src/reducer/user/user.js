import {extend} from "../../utils.js";
import {authInfoAdapter} from "../../adapter.js";


const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authInfo: {},
  comment: {},
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  CHANGE_AUTH_INFO: `CHANGE_AUTH_INFO`,
  СHANGE_COMMENT: `СHANGE_COMMENT`,
};

const ActionCreator = {
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),

  changeAuthInfo: (authData) => ({
    type: ActionType.CHANGE_AUTH_INFO,
    payload: authData,
  }),

  changeComment: (dataComment) => ({
    type: ActionType.СHANGE_COMMENT,
    payload: dataComment,
  }),

};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(() => dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH)))
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.changeAuthInfo(authInfoAdapter(response.data)));
      });
  },

  sendComment: (commentData, hotelId) => (dispatch, getState, api) => {
    return api.post(`/comments/${hotelId}`, {
      comment: commentData.comment,
      rating: commentData.rating,
    })
      .then((response) => {
        dispatch(ActionCreator.changeComment(response.data));
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })
      .catch((err) =>{
        throw err;
      });
  },

};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionType.CHANGE_AUTH_INFO:
      return extend(state, {
        authInfo: action.payload,
      });
  }

  return state;
};


export {AuthorizationStatus, ActionType, ActionCreator, reducer, Operation};
