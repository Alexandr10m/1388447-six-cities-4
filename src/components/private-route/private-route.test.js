import React from "react";
import renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space.js";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {Router} from "react-router-dom";
import FavouritePage from "../favourite-page/favourite-page.jsx";
import history from "../../history.js";


const mockStore = configureStore([]);
const favourite = [];

describe(`Snapshot of PrivateRoute`, () => {
  it(`Should render correct SignIn component `, () => {
    const store = mockStore({
      [NameSpace.STATE]: {},
      [NameSpace.DATA]: {
        favourite: [],
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
        authInfo: {
          email: `iii`
        },
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <PrivateRoute
                render={()=>(<FavouritePage favouriteOffers={favourite}/>)}
                path={`/favourite`}
                exact={true}
                authorizationStatus={AuthorizationStatus.NO_AUTH}
              />
            </Router>
          </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Should render correct FavouritePage component `, () => {
    const store = mockStore({
      [NameSpace.STATE]: {},
      [NameSpace.DATA]: {
        favourite: [],
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
        authInfo: {
          email: `iii`
        },
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <PrivateRoute
                render={()=>(<FavouritePage favouriteOffers={favourite}/>)}
                path={`/favourite`}
                exact={true}
                authorizationStatus={AuthorizationStatus.AUTH}
              />
            </Router>
          </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
