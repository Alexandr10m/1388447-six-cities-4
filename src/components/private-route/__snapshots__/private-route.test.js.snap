// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of PrivateRoute Should render correct 1`] = `null`;

exports[`Snapshot of PrivateRoute Should render correct FavouritePage component  1`] = `null`;

exports[`Snapshot of PrivateRoute Should render correct SignIn component  1`] = `null`;
